features/cellular/framework/AT/AT_CellularNetwork.cpp

```diff
@@ -546,10 +546,20 @@ nsapi_error_t AT_CellularNetwork::set_registration(const char *plmn)
 
     if (!plmn) {
         tr_debug("Automatic network registration");
-        _at.cmd_start("AT+COPS=0");
+        uint8_t len=8;
+        uint8_t buf[8];
+        _at.cmd_start("AT+COPS?");
         _at.cmd_stop();
         _at.resp_start();
```

Comment:
You could do this also without need to declare buf and doing string compare:

_at.resp_start("+COPS:");
int mode = _at.read_int();
_at.resp_stop();

if (mode != 0) { ...

Reply:
Will check and update soon.